storage/client/client.go

```diff
@@ -55,6 +56,17 @@ func toString(b *BigInt) *string {
 	return &s
 }
 
+func runtimeFromCtx(ctx context.Context) string {
+	// Extract the runtime name. It's populated by a middleware based on the URL.
+	runtime, ok := ctx.Value(common.RuntimeContextKey).(string)
+	if !ok {
+		// We're being called from a non-runtime-specific endpoint.
+		// This shouldn't happen. Return a dummy value, let the caller deal with it.
+		return "__NO_RUNTIME__"
```

Comment:
Should this just panic as this is likely a developers mistake? e.g. calling this method from a non-runtime specific endpoint?

Reply:
It is a developers mistake for sure yeah. However, this is only used on the API side, i.e. when we're serving data from the DB. If we have a bug in a single endpoint handler, I'd rather produce an error log (which this __NO_RUNTIME__ value will probably eventually result in) than crash the entire API server.